### ISSUE TITLE
Can not train - images not found

### ISSUE THREAD

**Issue by palorajcan** [bmaltais/kohya_ss#981]
Hey guys, I am very new here and very new to the whole AI image-generating thing. So, forgive my noob question. I have installed kohya_ss according to this tutorial https://www.youtube.com/watch?v=j-So4VYTL98 and I am trying to learn to train. But something goes wrong because it seems like my images are not found. How to tackle this issue, please?

This is what I get when training model:

[Dataset 0]
loading image sizes.
0it [00:00, ?it/s]
make buckets
min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is set, because bucket reso is defined by image size automatically / bucket_no_upscaleが指定された場合は、bucketの解像度は画像サイズから自動計算されるため、min_bucket_resoとmax_bucket_resoは無視されます
number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）
D:\Stable Diffusion\Kohya\kohya_ss\venv\lib\site-packages\numpy\core\fromnumeric.py:3432: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
D:\Stable Diffusion\Kohya\kohya_ss\venv\lib\site-packages\numpy\core\_methods.py:190: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
mean ar error (without repeats): nan
No data found. Please verify arguments (train_data_dir must be the parent of folders with images) / 画像がありません。引数指定を確認してください（train_data_dirには画像があるフォルダではなく、画像があるフォルダの親フォルダを指定する必要があります）

and this is what I get when training lora:

10:39:48-624933 INFO     nVidia toolkit detected
10:39:49-632048 INFO     Torch 2.0.1+cu118
10:39:49-645048 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
10:39:49-646048 INFO     Torch detected GPU: NVIDIA GeForce RTX 3060 Ti VRAM 8192 Arch (8, 6) Cores 38
10:39:49-647048 INFO     Verifying requirements
10:39:51-004887 INFO     headless: False
10:39:51-006887 INFO     Load CSS...
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
10:41:12-322315 INFO     Start training LoRA Standard ...
10:41:12-328315 INFO     Total steps: 0
10:41:12-329314 INFO     Train batch size: 1
10:41:12-330314 INFO     Gradient accumulation steps: 1
10:41:12-331315 INFO     Epoch: 2
10:41:12-332315 INFO     Regulatization factor: 1
10:41:12-333315 INFO     max_train_steps (0 / 1 / 1 * 2 * 1) = 0
10:41:12-334315 INFO     stop_text_encoder_training = 0
10:41:12-335315 INFO     lr_warmup_steps = 0
10:41:12-336315 INFO     accelerate launch --num_cpu_threads_per_process=2 "train_network.py" --enable_bucket
                         --pretrained_model_name_or_path="D:/Stable
                         Diffusion/stable-diffusion-webui/models/Stable-diffusion/v1-5-pruned.safetensors"
                         --train_data_dir="D:/Stable Diffusion/Training/CB/images/20_CB" --resolution=512,512
                         --output_dir="D:/Stable Diffusion/Training/CB/model" --logging_dir="D:/Stable
                         Diffusion/Training/CB/log" --network_alpha="1" --save_model_as=safetensors
                         --network_module=networks.lora --text_encoder_lr=5e-05 --unet_lr=0.0001 --network_dim=8
                         --output_name="last" --lr_scheduler_num_cycles="2" --learning_rate="0.0001"
                         --lr_scheduler="cosine" --train_batch_size="1" --save_every_n_epochs="1"
                         --mixed_precision="fp16" --save_precision="fp16" --cache_latents --optimizer_type="AdamW8bit"
                         --max_data_loader_n_workers="0" --bucket_reso_steps=64 --xformers --bucket_no_upscale
[10:41:16] WARNING  The following values were not passed to `accelerate launch` and had defaults used      launch.py:890
                    instead:
                            `--num_processes` was set to a value of `1`
                            `--num_machines` was set to a value of `1`
                            `--mixed_precision` was set to a value of `'no'`
                            `--dynamo_backend` was set to a value of `'no'`
                    To avoid this warning pass in values for each of the problematic parameters or run
                    `accelerate config`.
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
prepare tokenizer
Using DreamBooth method.
prepare images.
0 train images with repeating.
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 1
  resolution: (512, 512)
  enable_bucket: True
  min_bucket_reso: 256
  max_bucket_reso: 1024
  bucket_reso_steps: 64
  bucket_no_upscale: True

[Dataset 0]
loading image sizes.
0it [00:00, ?it/s]
make buckets
min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is set, because bucket reso is defined by image size automatically / bucket_no_upscaleが指定された場合は、bucketの解像度は画像サイズから自動計算されるため、min_bucket_resoとmax_bucket_resoは無視されます
number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）
D:\Stable Diffusion\Kohya\kohya_ss\venv\lib\site-packages\numpy\core\fromnumeric.py:3432: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
D:\Stable Diffusion\Kohya\kohya_ss\venv\lib\site-packages\numpy\core\_methods.py:190: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
mean ar error (without repeats): nan
No data found. Please verify arguments (train_data_dir must be the parent of folders with images) / 画像がありません。引数指定を確認してください（train_data_dirには画像があるフォルダではなく、画像があるフォルダの親フォルダを指定する必要があります）

**Comment by ruiboy1919** [comment]
Think of [] as a folder name.

[test]-[1_image].
Instead of specifying the folder path of [1_image], specify the folder path of [test].

**Comment by bmaltais** [comment]
Download the latest release. It will check for proper folder structure and point you to the document that explain it in case it does not.

**Comment by palorajcan** [comment]
> 

Well, my folder for images is D:\Stable Diffusion\Training\CB\images\20_CB, so all images are in folder 20_CB. That is the exact path that I put in Kohya_ss GUI image folder. Images are jpgs with name structure "cb (number)". 

I tried to specify the path as D:\Stable Diffusion\Training\CB\images\ but it did not help. I am not very good with python, so I am doing something wrong apparently.

**Comment by palorajcan** [comment]
> Think of [] as a folder name.
> 
> [test]-[1_image]. Instead of specifying the folder path of [1_image], specify the folder path of [test].

Well, my folder for images is D:\Stable Diffusion\Training\CB\images\20_CB, so all images are in folder 20_CB. That is the exact path that I put in Kohya_ss GUI image folder. Images are jpgs with name structure "cb (number)".

I tried to specify the path as D:\Stable Diffusion\Training\CB\images\ but it did not help. I am not very good with python, so I am doing something wrong apparently.

**Comment by palorajcan** [comment]
> Download the latest release. It will check for proper folder structure and point you to the document that explain it in case it does not.

I have downloaded it yesterday evening, so I suppose it should be up to date. Am I wrong?

**Comment by bmaltais** [comment]
You are pointing the path at the wrong folder. Point it at D:\Stable Diffusion\Training\CB\images instead.

**Comment by palorajcan** [comment]
Got it running, thank you guys. Appreciate it.

**Comment by jammer699669** [comment]
well it did NOT work for me no matter how I tried it it never worked --- /home/james/HOAJACK2/kohya_ss/dataset/Hoa_Jack/class1/    and the files are named as j1.jpg and j1.text etc  ( yes not windows linux )  always the same error same error that he got and chatGPT is a moron wants me to patch files that do not even exist
